### PR TITLE
bqn: update

### PR DESCRIPTION
**Pull request by AndersonTorres**
Related to https://github.com/NixOS/nixpkgs/issues/311274

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/interpreters/bqn/cbqn/default.nix [hunk 1]
```diff
@@ -121,12 +127,18 @@ stdenv.mkDerivation rec {
...
   '';
 
-  meta = with lib; {
+  meta = {
     homepage = "https://github.com/dzaima/CBQN/";
     description = "BQN implementation in C";
-    license = licenses.gpl3Plus;
-    maintainers = with maintainers; [ AndersonTorres sternenseemann synthetica shnarazk detegr ];
-    platforms = platforms.all;
+    license = lib.licenses.gpl3Plus;
```
Comment (Detegr) [inline pkgs/development/interpreters/bqn/cbqn/default.nix:133]: CBQN has multiple licenses nowadays, see: https://github.com/dzaima/CBQN?tab=readme-ov-file#licensing

**Review by Detegr — APPROVED** [review]

File: pkgs/development/interpreters/bqn/cbqn/default.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/development/interpreters/bqn/cbqn/default.nix]: OK!

**Comment by matthiasbeyer** [comment]
Result of `nixpkgs-review pr 311543` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>bqn (mbqn)</li>
    <li>cbqn</li>
    <li>cbqn-replxx</li>
    <li>cbqn-standalone</li>
    <li>cbqn-standalone-replxx</li>
    <li>cbqn-standalone.dev</li>
    <li>cbqn-standalone.lib</li>
    <li>cbqn.dev</li>
    <li>cbqn.lib</li>
    <li>dbqn-native</li>
  </ul>
</details>

**Comment by AndersonTorres** [comment]
Another day, another update!

**Comment by AndersonTorres** [comment]
Error:

> cbqn-standalone> jit/nvm.c
> cbqn-standalone> utils/ryu.c
> cbqn-standalone> utils/utf.c
> cbqn-standalone> utils/hash.c
> cbqn-standalone> utils/file.c
> cbqn-standalone> utils/mut.c
> cbqn-standalone> utils/each.c
> cbqn-standalone> utils/bits.c
> cbqn-standalone> [0.0] genArithTables.bqn
> cbqn-standalone> /build/source/src/singeli/src/squeeze.singeli:10:
> cbqn-standalone> def anySNaN{M, x:T if eltype{T}==u64} = {
> cbqn-standalone> |          ^
> cbqn-standalone> |          params
> cbqn-standalone> define     lambda
> cbqn-standalone> Expected "}" but saw "if"
> cbqn-standalone> /build/source/src/singeli/src/cmp.singeli:19:
> cbqn-standalone> Error: Fork processing "squeeze.singeli" died
> cbqn-standalone> def pathAS{dst, len, T, op, x if issigned{T}} = {
> cbqn-standalone> |         ^
> cbqn-standalone> |         params
> cbqn-standalone> define    lambda
> cbqn-standalone> Expected "}" but saw "if"
> cbqn-standalone> /build/source/src/singeli/src/select.singeli:1:
> cbqn-standalone> if_inline (hasarch{'AVX2'}) {
> cbqn-standalone> /build/source/src/singeli/src/count.singeli:1:
> cbqn-standalone> ^
> cbqn-standalone> body
> cbqn-standalone> include './base'
> cbqn-standalone> Expected separator
> cbqn-standalone> ^
> cbqn-standalone> body
> cbqn-standalone> Expected separator
> cbqn-standalone> /build/source/src/singeli/src/slash.singeli:1:
> cbqn-standalone> include './base'
> cbqn-standalone> ^
> cbqn-standalone> body
> cbqn-standalone> Expected separator
> cbqn-standalone> Error: Fork processing "cmp.singeli" died
> cbqn-standalone> /build/source/src/singeli/src/bins.singeli:1:
> cbqn-standalone> include './base'
> cbqn-standalone> ^
> cbqn-standalone> body
> cbqn-standalone> Expected separator
> cbqn-standalone> Error: Fork processing "select.singeli" died
> cbqn-standalone> Error: Fork processing "slash.singeli" died
> cbqn-standalone> Error: Fork processing "bins.singeli" died
> cbqn-standalone> /build/source/src/singeli/src/replicate.singeli:25:
> cbqn-standalone>   def scan{ptr, len if width{T}<=32} = {
> cbqn-standalone>   |       ^
> cbqn-standalone>   |       params
> cbqn-standalone>   define  lambda
> cbqn-standalone> Expected "}" but saw "if"
> cbqn-standalone> Error: Fork processing "replicate.singeli" died
> cbqn-standalone> /build/source/src/singeli/src/base.singeli:27:
> cbqn-standalone> def andnot{a, b if anyNum{a} and anyNum{b}} = a & ~b
> cbqn-standalone> |         ^
> cbqn-standalone> |         params
> cbqn-standalone> define    lambda
> cbqn-standalone> Expected "}" but saw "if"
> cbqn-standalone> Error: Fork processing "monarith.singeli" died
> cbqn-standalone> /build/source/src/singeli/src/dyarith.singeli:9:
> cbqn-standalone> def rootty{T if isprim{T}} = T
> cbqn-standalone> |         ^
> cbqn-standalone> |         params
> cbqn-standalone> define    lambda
> cbqn-standalone> Expected "}" but saw "if"
> cbqn-standalone> Error: Fork processing "dyarith.singeli" died
> cbqn-standalone> [0.0] versionInfo.c
> cbqn-standalone> [0.1] group.c
> cbqn-standalone> [0.5] sfns.c
> cbqn-standalone> 4/64 in 0.5s; failed to build
> cbqn-standalone> make[1]: *** [makefile:85: to-bqn-build] Error 1
> cbqn-standalone> make: *** [makefile:7: o3] Error 2
> error: builder for '/nix/store/9m6vz43limd3rikij5d57glqjb7by9n4-cbqn-standalone-0.7.0.drv' failed with exit code 2;
>        last 25 log lines:
>        >
>        > Expected "}" but saw "if"
>        > Error: Fork processing "replicate.singeli" died
>        > /build/source/src/singeli/src/base.singeli:27:
>        > def andnot{a, b if anyNum{a} and anyNum{b}} = a & ~b
>        > |         ^
>        > |         params
>        > define    lambda
>        >
>        > Expected "}" but saw "if"
>        > Error: Fork processing "monarith.singeli" died
>        > /build/source/src/singeli/src/dyarith.singeli:9:
>        > def rootty{T if isprim{T}} = T
>        > |         ^
>        > |         params
>        > define    lambda
>        >
>        > Expected "}" but saw "if"
>        > Error: Fork processing "dyarith.singeli" died
>        > [0.0] versionInfo.c
>        > [0.1] group.c
>        > [0.5] sfns.c
>        > 4/64 in 0.5s; failed to build
>        > make[1]: *** [makefile:85: to-bqn-build] Error 1
>        > make: *** [makefile:7: o3] Error 2
>        For full logs, run 'nix log /nix/store/9m6vz43limd3rikij5d57glqjb7by9n4-cbqn-standalone-0.7.0.drv'.
> error: 1 dependencies of derivation '/nix/store/5xbxgcy7kwgx6schsx630b52l07b0h6l-cbqn-0.7.0.drv' failed to build
> error: 1 dependencies of derivation '/nix/store/3bsknzghfb4fy4ldilaj6x1rg7nqbi94-cbqn-0.7.0.drv' failed to build
> 

0.7.0 is no way.

**Comment by dzaima** [comment]
@AndersonTorres that error would be from not having updated Singeli, I think. Between 0.6.0 and 0.7.0 it was updated, but the custom override here presumably wasn't.

**Comment by AndersonTorres** [comment]
Good to know. I will take this into account, now I am refactoring the mess.

**Comment by AndersonTorres** [comment]
Related: https://github.com/NixOS/nixpkgs/pull/314614

**Review by nicoonoclaste — APPROVED** [review]
Looks OK, can merge after `nixpkgs-review`

**Pull request merged**